### PR TITLE
v0.0.1: stable enough to be used as ktree or baby groot

### PR DESCRIPTION
**Pull request by ameydev**
You might wanna check --version command @arush-sal

### CODE REVIEW

**Review by arush-sal — CHANGES_REQUESTED** [review]
`bin/groot-0.1` should be removed. We rather should have a release when the PR is merged

File: cmd/show.gol [hunk 1]
```diff
@@ -0,0 +1,52 @@
+// /*
```
Comment (arush-sal) [inline cmd/show.gol]: This file needs to be removed.

File: kmap/mapper.go [hunk 2]
```diff
@@ -0,0 +1,387 @@
...
+	"k8s.io/client-go/kubernetes"
+)
+
+var rPool *ResourcePool
+var groot *Resource
+var indentationCount int = 0
+var indentation string
+
+func FindThemAll(clientset *kubernetes.Clientset, namespace *string) error {
+	// var kinds = "Pods,ComponentStatuses,ConfigMaps,Endpoints,LimitRanges,Namespaces,PersistentVolumes,PersistentVolumeClaims,PodTemplates,ResourceQuotas,Secrets,Services,ServiceAccounts,DaemonSets,Deployments,ReplicaSets,StatefulSets"
```
Comment (arush-sal) [inline kmap/mapper.go]: remove the unnecessary comments.

File: kmap/mapper.go [hunk 3]
```diff
@@ -0,0 +1,387 @@
...
+
+	"k8s.io/client-go/kubernetes"
+)
+
+var rPool *ResourcePool
+var groot *Resource
+var indentationCount int = 0
+var indentation string
+
+func FindThemAll(clientset *kubernetes.Clientset, namespace *string) error {
```
Comment (arush-sal) [inline kmap/mapper.go:17]: Exported functions should have a comment explaining what it is intended to do. `Lint the code`.

File: kmap/mapper.go [hunk 4]
```diff
@@ -0,0 +1,387 @@
...
+	groot = resourceNamespace
+	groot = mapThemAll(groot, rPool)
+	printTree(groot)
+	return nil
+}
+
+func mapThemAll(groot *Resource, rPool *ResourcePool) *Resource {
+	resources := rPool.resources
+	for _, resource := range resources {
+		// fmt.Println(resource.kind, " - ", resource.name, " Status: ", resource.status)
```
Comment (arush-sal) [inline kmap/mapper.go]: remove line

File: kmap/mapper.go [hunk 5]
```diff
@@ -0,0 +1,387 @@
...
+						pod.children = append(pod.children, resource)
+						resource.parent = pod
+						resource.hasParent = true
+						break
+					}
+				}
+				pvc := volume.VolumeSource.PersistentVolumeClaim
+				if pvc != nil && !resource.hasParent {
+					if volume.VolumeSource.PersistentVolumeClaim.ClaimName == resource.name {
+						// pod.children = append(pod.children, resource)
```
Comment (arush-sal) [inline kmap/mapper.go]: remove line

File: kmap/mapper.go [hunk 6]
```diff
@@ -0,0 +1,387 @@
...
+						resource.parent = pod
+						resource.hasParent = true
+						break
+					}
+				}
+			}
+
+		}
+	}
+	// groot.children = append(groot.children, pod)
```
Comment (arush-sal) [inline kmap/mapper.go]: remove line

File: kmap/mapper.go [hunk 7]
```diff
@@ -0,0 +1,387 @@
...
+	return pod, rPool
+}
+
+func findDeployChildren(deploy *Resource, rPool *ResourcePool) (*Resource, *ResourcePool) {
+
+	for _, resource := range rPool.resources {
+		// var isMapped bool
```
Comment (arush-sal) [inline kmap/mapper.go:157]: remove line

File: kmap/mapper.go [hunk 8]
```diff
@@ -0,0 +1,387 @@
...
+
+		if resource.kind == "Replicaset" || resource.kind == "PodTemplate" {
+			// fmt.Faddf(w, "%v\t%v\t%v\t%v\t%v\n", deployment.Name, deployment.Status.ReadyReplicas, "", deployment.Status.AvailableReplicas, "")
```
Comment (arush-sal) [inline kmap/mapper.go:160]: remove line

File: kmap/mapper.go [hunk 9]
```diff
@@ -0,0 +1,387 @@
...
+			selector := deploy.selector
+			if selector != nil && !resource.hasParent {
+				// filterPodsWithLabel(pods, selector)
```
Comment (arush-sal) [inline kmap/mapper.go:163]: remove line

File: kmap/mapper.go [hunk 10]
```diff
@@ -0,0 +1,387 @@
...
+			}
+
+		}
+	}
+
+	return StatefulSet, rPool
+}
+
+func findServiceChildren(service *Resource, rPool *ResourcePool) (*Resource, *ResourcePool) {
+	// var uPool *ResourcePool
```
Comment (arush-sal) [inline kmap/mapper.go:254]: remove line

File: kmap/mapper.go [hunk 11]
```diff
@@ -0,0 +1,387 @@
...
+}
+
+func findPVCChildren(pvc *Resource, rPool *ResourcePool) (*Resource, *ResourcePool) {
+
+	for _, resource := range rPool.resources {
+
+		if resource.kind == "PersistentVolume" && !resource.hasParent {
+
+			if pvc.info["Volume"] == resource.name {
+				// resource, uPool := findPodChildren(resource, rPool)
```
Comment (arush-sal) [inline kmap/mapper.go:335]: remove line

File: kmap/mapper.go [hunk 12]
```diff
@@ -0,0 +1,387 @@
...
+				pvc.children = append(pvc.children, resource)
+				resource.parent = pvc
+				resource.hasParent = true
+				break
+				// rPool = uPool
```
Comment (arush-sal) [inline kmap/mapper.go:340]: remove line

File: kmap/resources.go [hunk 13]
```diff
@@ -0,0 +1,388 @@
...
+}
+
+type ResourcePool struct {
+	resources []*Resource
+}
+
+func (r *ResourcePool) addToResourcePool(resource *Resource) {
+
+	r.resources = append(r.resources, resource)
+	// return pool
```
Comment (arush-sal) [inline kmap/resources.go]: remove line

File: kmap/resources.go [hunk 14]
```diff
@@ -0,0 +1,388 @@
...
+			blankResource.status = Int32toString(sset.Status.ReadyReplicas) + "/" + Int32toString(sset.Status.CurrentReplicas)
+			blankResource.Labels = sset.Labels
+			blankResource.selector = sset.Spec.Selector.MatchLabels
+			rPool.addToResourcePool(&blankResource)
+		}
+	}
+	return rPool
+}
+
+func Int32toString(n int32) string {
```
Comment (arush-sal) [inline kmap/resources.go:363]: add comment

Comment (arush-sal) [reply kmap/resources.go:363]: and instead of this discrete function use `strconv.Itoa()`

File: cmd/groot.go [hunk 15]
```diff
@@ -0,0 +1,147 @@
...
+	"os"
+
+	"github.com/spf13/cobra"
+
+	"os/user"
+
+	// Load all known auth plugins
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+
+	// "github.com/ameydev/groot/ksearch"
```
Comment (arush-sal) [inline cmd/groot.go]: remove line

File: cmd/groot.go [hunk 16]
```diff
@@ -0,0 +1,147 @@
...
+	"github.com/ameydev/groot/kmap"
+	homedir "github.com/mitchellh/go-homedir"
+	"github.com/spf13/viper"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// var cfgFile string
```
Comment (arush-sal) [inline cmd/groot.go]: remove line

File: cmd/groot.go [hunk 17]
```diff
@@ -0,0 +1,147 @@
...
+	if err != nil {
+		return err
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+
+	// fmt.Println("Namespace : " + c.Namespace + " \n ")
```
Comment (arush-sal) [inline cmd/groot.go]: remove line

File: cmd/groot.go [hunk 18]
```diff
@@ -0,0 +1,147 @@
...
+	// ksearch.SearchResources(clientset, &c.Namespace)
```
Comment (arush-sal) [inline cmd/groot.go]: remove line

File: cmd/groot.go [hunk 19]
```diff
@@ -0,0 +1,147 @@
...
+	kmap.FindThemAll(clientset, &c.Namespace)
+
+	return nil
+}
+
+func getKubeConfig() string {
```
Comment (arush-sal) [inline cmd/groot.go:128]: what about `in-cluster` config?

**Review by arush-sal — CHANGES_REQUESTED** [review]
Overall run the code through `go fmt` and `go vet`, clean up the code and fix all the linting issues.

File: kmap/mapper.go (hunk 3 above)
Comment (ghost) [reply kmap/mapper.go:17]: Id

> Exported functions should have a comment explaining what it is intended to do. `Lint the code`.

Added a comment about the function

**Comment by testuser-ops** [comment]
> `bin/groot-0.1` should be removed. We rather should have a release when the PR is merged

done. Removed bin

File: cmd/groot.go (hunk 19 above)
Comment (ameydev) [reply cmd/groot.go:128]: > what about `in-cluster` config?

Sorry I did not get what you are trying to say

Comment (arush-sal) [reply cmd/groot.go:128]: nvm let it be.

File: kmap/mapper.go (hunk 3 above)
Comment (arush-sal) [reply kmap/mapper.go:17]: Lint the code! the comment should be of type `// FindThemAll does stuff` and why are you using `testuser-ops` to comment here?

File: cmd/groot.go [hunk 20]
```diff
@@ -23,10 +23,8 @@ import (
 
 	"os/user"
 
-	// Load all known auth plugins
```
Comment (arush-sal) [inline cmd/groot.go]: this was a valid comment :man_facepalming:

File: kmap/resources.go (hunk 13 above)
Comment (ameydev) [reply kmap/resources.go]: removed the line..

File: kmap/resources.go (hunk 14 above)
Comment (ameydev) [reply kmap/resources.go:363]: I read somewhere that the descrete function is the fastest way :p thats why I used it though

Comment (arush-sal) [reply kmap/resources.go:363]: If you are good and okay with it then I don't have any issue.

**Review by arush-sal — APPROVED** [review]

File: kmap/resources.go (hunk 14 above)
Comment (ameydev) [reply kmap/resources.go:363]: It’s a temporary change. I think We should fetch status of children nodes and decide the status of parent node afterwards, so may not need this method

**Pull request merged**